### ISSUE TITLE
[Messenger] flatternException->getTraceAsString() returns null sometimes, which causes exceptions

### ISSUE THREAD

**Issue by wtorsi** [symfony/symfony#32487]
4.3.2

When `$flattenException` is built from `$lastRedeliveryStamp` in `Symfony\Component\Messenger\Command\AbstractFailedMessagesCommand:76` it may contains `null` in `traceAsString`. Which causes exceptions in  `Symfony\Component\Messenger\Command\AbstractFailedMessagesCommand:103`, because SymfonyStyle does not like nulls as messages.

```
        $flattenException = null === $lastRedeliveryStamp ? null : $lastRedeliveryStamp->getFlattenException();
```
```
            $io->title('Exception:');
            $io->writeln(null === $flattenException ? '(no data)' : $flattenException->getTraceAsString());
```

Actually, I don't know how to simply reproduce this exception. 
I got this exception, trying sending `Email` through `Mailer`, while generating body in `Email`. (`A message must have a text and/or an HTML part.`)

As a solution i could recommend the following:
1. Add strong (?string) type-hinting for `Symfony\Component\Debug\Exception\FlattenException::getTraceAsString`
2. Do check not only for `null === $flattenException` in Symfony\Component\Messenger\Command\AbstractFailedMessagesCommand:103`, but either check `null === $flattenException->getTraceAsString()`

**Comment by wtorsi** [comment]
Ah, the problem occurs, while serializing `TemplatedEmail` before sending it into the bus in `Mailer`. :)
```
    public function send(RawMessage $message, SmtpEnvelope $envelope = null): void
    {
        if (null === $this->bus) {
            $this->transport->send($message, $envelope);

            return;
        }

        $this->bus->dispatch(new SendEmailMessage($message, $envelope));
    }
``` 

Serializer is trying to `getBody` of `TemplatedEmail`, but the body at this moment is empty, no $html and $text parameters were set yet. So it's another small issue in `Mailer`.

**Comment by fabpot** [comment]
Should be fixed by #37847
